### PR TITLE
Implement retry for WebSocket client

### PR DESCRIPTION
**Pull request by Bhashinee**
## Purpose
Implement retry for WebSocket client. 
Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/1820, https://github.com/ballerina-platform/ballerina-standard-library/issues/1820

## Examples

Can enable retry using the following config:
```ballerina
websocket:Client wsClient = check new("ws://localhost:21078/websocket", { 
                                retryConfig: { 
                                    maxCount: 10,
                                    interval: 1,
                                    backOffFactor: 1.0,
                                    maxWaitInterval: 30
                               } 
                      }
             );
```

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests

### CODE REVIEW

File: ballerina/websocket_sync_client.bal [hunk 1]
```diff
@@ -253,13 +255,28 @@ public type CommonClientConfiguration record {|
...
+# Retry configurations for WebSocket.
+#
+# + maxCount - The maximum number of retry attempts. If the count is zero, the client will retry indefinitely.
+# + interval - The number of seconds to delay before attempting to reconnect.
+# + backOffFactor - The rate of increase of the reconnect delay. Allows reconnect attempts to back off when problems
+#                persist.
+# + maxWaitInterval - Maximum time of the retry interval in seconds.
+public type WebSocketRetryConfig record {|
+    int maxCount = 0;
+    int interval = 1;
```
Comment (shafreenAnfar) [inline ballerina/websocket_sync_client.bal:275]: Interval should be in decimal type

File: ballerina/websocket_sync_client.bal [hunk 2]
```diff
@@ -253,13 +255,28 @@ public type CommonClientConfiguration record {|
...
+    float backOffFactor = 1.0;
+    int maxWaitInterval = 30;
```
Comment (shafreenAnfar) [inline ballerina/websocket_sync_client.bal:277]: This should be decimal too. 

We use decimal in seconds for intervals in standard library.

File: ballerina/websocket_sync_client.bal (hunk 1 above)
Comment (Bhashinee) [reply ballerina/websocket_sync_client.bal]: Fixed with https://github.com/ballerina-platform/module-ballerina-websocket/pull/303/commits/6db591d8501ae3b57c3b79d2b078b3e0e5685bd5

File: ballerina/websocket_sync_client.bal (hunk 2 above)
Comment (Bhashinee) [reply ballerina/websocket_sync_client.bal]: Fixed with https://github.com/ballerina-platform/module-ballerina-websocket/pull/303/commits/6db591d8501ae3b57c3b79d2b078b3e0e5685bd5

**Review by shafreenAnfar — APPROVED** [review]

**Pull request merged**